### DISCUSSION TITLE
DataGrid Virtualize not applying paging

### DISCUSSION THREAD

**Discussion by leetheman** [microsoft/fluentui-blazor#762] [Q&A]
Hi there, I am trying to use the Virtualize feature, however it's not respecting the "Paging" element, so it's trying to pull the entire dataset from my IQueryable after skipping.  Is there a way to set a page size as well for virtualize, otherwise it doens't really work?

e.g. I have 400k records, using Paging it works perfectyl with Iqueryable.

but when I use virtualize, the query is SELECT * FROM tbl OFFSET etc.. instead of SELECT TOP X FROM tbl OFFSET, so even though its' skipping a few records it's still fetching everything.

I believe Telerik have cracked it in terms of allowing us to have Virtualize use the page size parameter to only fetch X items offset by a certain amount (we can tweak X in order to ensure enough overflow records)

**Comment by vnbaaij** [comment]
I'm not familiar with what Telerik has done there but to me Virtualize and Paging are mutually exclusive. 

You can be in control of what data gets queried. See the explanation at https://www.fluentui-blazor.net/DataGrid#remotedata and also https://aspnet.github.io/quickgridsamples/datasources (our FluentDataGrid is just a QuickGrid in disguise)
